students/douglas_klos/session5/mailroom3.py

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/env python3
+""" Mailroom V4.0, now with more error trapping """
+
+# Douglas Klos
+# February 15th, 2019
+# Python 210, Session 5, Mailroom v3
+# mailroom3.py
+
+import os
+import datetime
+
+mailroom_db = {'Douglas': [5000, 2000],
+               'Maggie': [2222, 3333, 4444],
+               'Light Yagami': [124, 8975],
+               'Makise Kurisu': [235987],
+               'Youjo Senki': [13498.00, 9876, 1234],
+               'Motoko Kusanagi': [57892, 239857, 87265],
+               'Jo': [8814, 2320]}
+
+MAIN_PROMPT = ('\nWelcome to the Mail Room\n'
+               'Please choose from the following options:\n'
+               '1: Send a Thank You\n'
+               '2: Write thank you files\n'
+               '3: Create a report\n'
+               '4: Add or Remove a Donor or Donation\n'
+               'p: Print database\n'
+               'q: Quit\n'
+               '>>> ')
+
+ADD_REMOVE_PROMPT = ('\n1: Add donation\n'
+                     '2: Remove donation\n'
+                     '3: Add donor\n'
+                     '4: Remove donor\n'
+                     'p: Print database\n'
+                     'q: Return to main menu\n'
+                     '>>> ')
+
+THANK_YOU_PROMPT = ('\nPlease enter one of the following:\n'
+                    '<Name>: Name of person to send a thank you note to\n'
+                    'list: Display donors in the database\n'
+                    'q: Return to main menu\n'
+                    '>>> ')
+
+THANK_YOU_NOTE = ('\nDear {}:\n'
+                  '\tThank you for your very kind donation of ${:,.2f}.\n'
+                  '\tIt will be put to very good use.\n'
+                  '\t\tSincerely,\n'
+                  '\t\t\tThe Team')
+
+THANK_YOU_LETTER = ('Dear {}:\n'
+                    '\tThank you for your most recenet donation of ${:,.2f}.\n'
+                    '\tYour total generosity towards us is ${:,.2f}.\n'
+                    '\tIt will be put to very good use.\n'
+                    '\t\tSincerely,\n'
+                    '\t\t\tThe Team\n')
+
+
+def display_database():
+    """ Displays all donors and their donations """
+
+    print()
+
+    for key in mailroom_db:
+        print(f'{key:>24} : {mailroom_db[key]:}')
```

Comment:
you could use a comp here.  I know the text says something about dont' use for prints, but yes, there isn't really a spot for this in this assignment except to recognize where they COULD be used.  Typically we only use comps for speed as they are faster than for loops.